pkg/await/apps_deployment.go

```diff
@@ -416,10 +417,8 @@ func (dia *deploymentInitAwaiter) processDeploymentEvent(event watch.Event) {
 			continue
 		}
 
-		if progressingStatusUnavailable {
-			// Since we can't tell for sure from this version of the API, mark as available.
-			dia.replicaSetAvailable = true
-		} else if condition["type"] == "Progressing" {
+		if condition["type"] == "Progressing" {
+		    progressingStatusUnavailable = false
```

Comment:
I'm confused about why this is correct, and why the previous way of doing this is incorrect. Also, if the controller is slow to add the progressing condition initially, we'll just mark the deployment as available, won't we? That doesn't seem right.

Reply:
The problem I'm running into is that it doesn't seem to be possible to tell if the Progressing status will ever exist for a Deployment. If it was created with the `extensions/v1beta1` API, it will not include that status, even if you request the `apps/v1` version. I missed that nuance previously, so explicitly checking the apiVersion on the received object doesn't do any good.

I think we need to reexamine the await logic for Deployment, and possibly avoid using this status field to make a determination since it doesn't work in general.

Reply:
We can tell based on the inputs what API version we was used to create this resource, though, right?